File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_dict = {}

        for index, num in enumerate(nums):
            complement = target - num

            if complement in num_dict:
                return [num_dict[complement], index]

            num_dict[num] = index

        return None  # Explicitly return None if no solution is found.

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

// brute force solution
var twoSum = function(nums, target) {
    
    let obj = {}
    
    for ( let i = 0 ; i < nums.length ; i++ ) {
        let diff = target - nums[i]
        if(diff in obj){
            return [ obj[diff] , i ]
        }
        obj[ nums[i] ] = i
    }
};